### PR TITLE
feat(download) Introduce downloadTaskError type

### PR DESCRIPTION
**Pull request by holyhope**
Introduce the download task error type, such as [the file system error type](https://github.com/NikolaLohinski/free-go/blob/main/types/filesystem.go#L45-L77).

### CODE REVIEW

File: types/downloads.go [hunk 1]
```diff
@@ -26,6 +26,16 @@ const (
...
 )
 
+type downloadTaskError string
+
+const (
+	DownloadTaskErrorNone             downloadTaskError = "none"          // No error
+	DownloadTaskErrorHashDownloadDisk downloadTaskError = "hash_download" // Error downloading the hash file
+	DownloadTaskError4XX              downloadTaskError = "http_4xx"      // Error 4xx
+	DownloadTaskErrorNotFound         downloadTaskError = "not_found"     // Error 404
+	// TODO: Add more error codes
```
Comment (NikolaLohinski) [inline types/downloads.go]: What's with this todo?

Are the error code strings not documented and you've been reverse engineering them ? (If so kuddos)

**Review by NikolaLohinski — CHANGES_REQUESTED** [review]

File: types/downloads.go (hunk 1 above)
Comment (holyhope) [reply types/downloads.go]: You are right, I added all the documented error code, and put my retro-engineered error codes in a dedicated section.

Wdyt about this new list?

Comment (NikolaLohinski) [reply types/downloads.go]: Sounds good to me, thanks for doin the heavy work, I know it's can be painful. Would you mind updating the TODO to mention that it's undocumented and that those errors were manually gathered ? (something of sorts, the wording is not super important)

File: types/downloads.go [hunk 2]
```diff
@@ -26,6 +26,34 @@ const (
...
+	DownloadTaskErrorInvalidAddress    downloadTaskError = "invalid_address"      // Invalid peer address
+	DownloadTaskErrorPortConflict      downloadTaskError = "port_conflict"        // Port conflict when setting config
+	DownloadTaskErrorInvalidPriority   downloadTaskError = "invalid_priority"     // Invalid priority
+	DownloadTaskErrorInternalError     downloadTaskError = "internal_error"       // Internal error
+	DownloadTaskErrorCtxFileError      downloadTaskError = "ctx_file_error"       // Failed to initialize task context file (need to check disk)
+	DownloadTaskErrorExists            downloadTaskError = "exists"               // Same task already exists
+	DownloadTaskErrorPortOutsideRange  downloadTaskError = "port_outside_range"   // Incoming port is not available for this customer (see ConnectionStatus ipv4_port_range)
+
+	// Retro-engineered error codes
+	DownloadTaskErrorNone             downloadTaskError = "none"              // No error
```
Comment (holyhope) [inline types/downloads.go]:
```suggestion
	// Undocumented and reverse engineered error codes
	DownloadTaskErrorNone             downloadTaskError = "none"              // No error
```

File: types/downloads.go (hunk 1 above)
Comment (holyhope) [reply types/downloads.go]: Something like that?
https://github.com/NikolaLohinski/free-go/pull/5/files#r1803688563

Comment (NikolaLohinski) [reply types/downloads.go]: Perfectomundo 👌

**Review by NikolaLohinski — APPROVED** [review]

**Pull request merged**